### PR TITLE
Adding object replication to list blob response

### PR DESCRIPTION
**Pull request by amnguye**
Related to PR #12609 . This PR contains the API View comment changes.

Object Replication for List Blobs Item Properties

This PR enables users to retrieve a List of `ObjectReplicationPolicy`(s) that contains the details of the Object Replication Metadata (OrMetadata) applied to the source blob when calling List/Get Blobs.

For information on how `ObjectReplicationPolicy`, `ObjectReplicationRules`, and `ReplicationStatus` is formatted see PR #12740

I also added Playback Only to all the OR tests

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs [hunk 1]
```diff
@@ -594,7 +594,7 @@ public async Task DownloadAsync_VersionId()
             Assert.IsNotNull(response.Value.Details.VersionId);
         }
 
-        // TODO: Recorded Only
+        [PlaybackOnly("Object Replication policies is only enabled on certain storage accounts")]
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:597]: Did the PlaybackOnly attribute work correctly?

**Review by seanmcc-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:597]: It actually doesn't. I just tested this now. Should I take it off then?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:597]: Oh no!  Did it have any affect?

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:597]: no :confused:

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs:597]: Ok let’s follow up on Monday, I’ll need to revisit my PR for the test attribute.

**Pull request merged**